routers/web/repo/repo.go

```diff
@@ -334,6 +339,30 @@ func Action(ctx *context.Context) {
 		return
 	}
 
+	switch ctx.Params(":action") {
+	case "watch", "unwatch":
+		ctx.Data["IsWatchingRepo"] = repo_model.IsWatching(ctx, ctx.Doer.ID, ctx.Repo.Repository.ID)
+	case "star", "unstar":
+		ctx.Data["IsStaringRepo"] = repo_model.IsStaring(ctx, ctx.Doer.ID, ctx.Repo.Repository.ID)
+	}
+
+	switch ctx.Params(":action") {
+	case "watch", "unwatch", "star", "unstar":
+		ctx.Data["Repository"], err = repo_model.GetRepositoryByName(ctx, ctx.Repo.Repository.OwnerID, ctx.Repo.Repository.Name)
```

Comment:
Why not directly
```suggestion
		ctx.Data["Repository"] = ctx.Repo.Repository
```

Reply:
Because the `NumStars`/`NumWatches` changed

Reply:
Maybe we can put a comment here.

Reply:
Done https://github.com/go-gitea/gitea/pull/29001/commits/888553695784e537c8f4435c83280f0c28fe7472